src/test/ui/rfc-2497-if-let-chains/feature-gate.rs

```diff
@@ -0,0 +1,102 @@
+// gate-test-let_chains
+
+// Here we test feature gating for ´let_chains`.
+// See `disallowed-positions.rs` for the grammar
+// defining the language for gated allowed positions.
+
+#![allow(irrefutable_let_patterns)]
+
+use std::ops::Range;
+
+fn _if() {
+    if let 0 = 1 {} // Stable!
+
+    if (let 0 = 1) {}
+    //~^ ERROR `let` expressions in this position are experimental [E0658]
+    //~| ERROR `let` expressions only supported in `if`
```

Comment:
Why is this error happening?
The expression looks correct - `let` is inside `if` and only parentheses are between.

Reply:
This is due to https://github.com/rust-lang/rust/pull/60861/files#diff-64b696b0ef6ad44140e973801ed82b25R4120:
https://github.com/rust-lang/rust/blob/a452e0fb8043dc9a3d83df66d36d1ccdfb5285d1/src/librustc/hir/lowering.rs#L4118-L4124

It is a stop-gap solution so that this PR doesn't produce ICEs due to not having implemented the feature in `lowering.rs`. It will get removed in a subsequent PR.

Reply:
I see now, this PR doesn't implement the lowering part for the new cases, only for the old ones (`if let`/`while let`).